src/upgrades/mongodb_upgrade.py

```diff
@@ -189,6 +190,7 @@ def _on_force_upgrade_action(self, event: ActionEvent) -> None:
         logger.debug("Forcing upgrade")
         event.log(f"Forcefully upgrading {self.charm.unit.name}")
         self._upgrade.upgrade_unit(charm=self.charm)
+        self.charm.version_checker.set_version_across_all_relations()
```

Comment:
QQ: Is this needed here?

I don't believe it is, but I could very well be wrong. 

Reasoning: `_on_pre_upgrade_check_action` actions in two cases:
1. before the user runs `juju refresh ...`
2. when the user runs `juju refresh ...`

In case 1, no version has changed so its not necessary share across all relations
In case 2, I believe your change on line `135` is sufficent

Reply:
> I'm guessing you mean _on_force_upgrade_action but in any case.

Oops, yes you are right, I was too speedy in my review thank you for catching this. 

Some context is that `_on_force_upgrade_action ` can be run on *any* unit (i.e. non-leader units) and `set_version_across_all_relations ` **must** be ran on the leader unit. So I don't think this is the right place for it.

What we can do instead, is change our `def _on_upgrade_charm(self, _):` function to switch up our order of operations. namely modify it to:
```
            self.charm.version_checker.set_version_across_all_relations()
            self._reconcile_upgrade()
```
this way we set the version before actually begging to reconcile. This way errors in `_reconcile_upgrade` will not block `set_version_across_all_relations`.

Some Juju knowledge for you:
If there is a python error during an event handler, that unit goes into the `error` state, and will keep calling that hook until it resolves itself. So we can trust once the charm goes out of `error` state that `set_version_across_all_relations` will eventually be called.

Now for your second point:
> Maybe we should also have tests for force upgrade action?

If you'd like to implement this, I would be happy to support you on it, but we would have to make a separate JIRA ticket. I imagine the timing of simulating a failure in a timely matter (such that pre-upgrade check succeeds, but post-upgrade check fails) would be challenging.

Reply:
Ok perfect, I'll do this then.
For the additional test it needs to be an extra ticket but it would be nice if it helps having a framework to test for such errors.